src/overlays/actors/ovl_Bg_Sinkai_Kabe/z_bg_sinkai_kabe.c

```diff
@@ -2,9 +2,15 @@
  * File: z_bg_sinkai_kabe.c
  * Overlay: ovl_Bg_Sinkai_Kabe
  * Description: Manages the Deep Pythons and Seahorse in Pinnacle Rock
```

Comment:
Is the seahorse part correct?

Reply:
This is definitely correct, though I could probably phrase it better. Let's assume the player does the following:
- Rescue the Seahorse from the Fisherman
- Go to Pinnacle Rock and follow the Seahorse through the sign maze
- Defeat all eight Deep Pythons to spawn the Seahorse's friend
- Leave the area without talking to the Seahorse
- Return to Pinnacle Rock and blindly navigate the sign maze

Assuming you did that frankly crazy sequence of steps, you'd expect to see the Seahorse again in the same burrow he was in before, right? So that's what Sinkai_Kabe does; it spawns the Seahorse in the scenario where you leave and re-enter Pinnacle Rock. Let me try to explain that better within the file.